docs/en-us/sidecarset.md

```diff
@@ -278,6 +278,73 @@ spec:
 ```
 **Note: If you use Scatter, it is recommended to set only a pair of key-values for scatter. It will be easier to understand.**
 
+### Hot Upgrade Sidecar
+**FEATURE STATE:** Kruise v0.9.0
+
+SidecarSet's in-place upgrade will stop the container of old version first and then create the container of new version. Such method is more suitable for stateless sidecar containers, e.g. logging collector.
+
+But for many stateful sidecar containers, e.g. Istio Envoy, this upgrade method is problematic. Envoy, as a proxy container in the Pod, proxies all the traffic, and if restarted directly, the availability of service is affected. So we provide a new solution for such stateful sidecar container upgrade.
```

Comment:
Stateful sidecar containers is not well defined noun. Consider the following alternative:

SidecarSet's in-place upgrade will stop the container of old version first and then create the container of new version. Such method is more suitable for sidecar containers that cannot affects service availability, e.g. logging collector.

But for many proxy or runtime sidecar containers, e.g. Istio Envoy, this upgrade method is problematic. Envoy, as a proxy container in the Pod, proxies all the traffic, and if restarted directly, the availability of service is affected. Complex grace termination and coordination is required if one need to upgrade envoy sidecar independently of the application container. So we provide a new solution for such sidecar container upgrade.